### PR TITLE
Wrapping up Infinispan

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring

### Description

This PR wrap up the Infinispan work for the moment.

It fixes up the configuration, adds the tenant cleanup, and contains the refactorings that had been intended for Infinispan 10.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

File: Makefile.env.mk [hunk 1]
```diff
@@ -51,10 +51,10 @@ IOT_HTTP_ADAPTER_IMAGE               ?= $(DOCKER_REGISTRY_PREFIX)$(DOCKER_ORG)/i
 IOT_MQTT_ADAPTER_IMAGE               ?= $(DOCKER_REGISTRY_PREFIX)$(DOCKER_ORG)/iot-mqtt-adapter:$(IMAGE_VERSION)
 IOT_LORAWAN_ADAPTER_IMAGE            ?= $(DOCKER_REGISTRY_PREFIX)$(DOCKER_ORG)/iot-lorawan-adapter:$(IMAGE_VERSION)
 IOT_SIGFOX_ADAPTER_IMAGE             ?= $(DOCKER_REGISTRY_PREFIX)$(DOCKER_ORG)/iot-sigfox-adapter:$(IMAGE_VERSION)
+IOT_TENANT_CLEANER_IMAGE             ?= $(DOCKER_REGISTRY_PREFIX)$(DOCKER_ORG)/iot-tenant-cleaner:$(IMAGE_VERSION)
```
Comment (famartinrh) [inline Makefile.env.mk]: can you add this image to the lists below IMAGE_ENV and IMAGE_LIST thanks

Comment (ctron) [reply Makefile.env.mk]: Thanks. I overlooked that. Fixed!

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

File: pkg/controller/consoleservice/consoleservice_controller.go [hunk 2]
```diff
@@ -10,11 +10,12 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"strconv"
```
Comment (jbtrystram) [inline pkg/controller/consoleservice/consoleservice_controller.go]: are not imports supposed to be in alphabetical order ?

File: templates/iot/examples/infinispan/manual/050-StatefulSet.yaml [hunk 3]
```diff
@@ -91,7 +91,7 @@ spec:
         volumeMounts:
         - mountPath: /opt/jboss/infinispan-server/standalone/configuration/custom
           name: infinispan-configuration
-        - mountPath: /opt/jboss/infinispan-server/standalone/data
+        - mountPath: /opt/infinispan/server/data/
```
Comment (jbtrystram) [inline templates/iot/examples/infinispan/manual/050-StatefulSet.yaml]: this looks like infinispan 10 image path

File: pkg/controller/iotproject/tenant_cleanup.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	iotv1alpha1 "github.com/enmasseproject/enmasse/pkg/apis/iot/v1alpha1"
+)
+
+var (
+	tenantCleanupNamespace uuid.UUID = uuid.MustParse("20c0db1c-ffc8-11e9-800c-c85b762e5a2c")
```
Comment (dejanb) [inline pkg/controller/iotproject/tenant_cleanup.go]: Any particular reason this is not a random value?

File: pkg/controller/consoleservice/consoleservice_controller.go (hunk 2 above)
Comment (ctron) [reply pkg/controller/consoleservice/consoleservice_controller.go]: I let IntelliJ handle that :D

File: templates/iot/examples/infinispan/manual/050-StatefulSet.yaml (hunk 3 above)
Comment (ctron) [reply templates/iot/examples/infinispan/manual/050-StatefulSet.yaml]: Yes, but I changed to config as well, that should work.

File: iot/iot-infinispan-base/src/main/java/io/enmasse/iot/infinispan/device/CredentialKey.java [hunk 5]
```diff
@@ -23,7 +22,6 @@
  *
  * See {@link CacheCredentialService CacheCredentialService} class.
  */
-@ProtoMessage
```
[the remark this answers is not in the document]
Comment (ctron) [reply iot/iot-infinispan-base/src/main/java/io/enmasse/iot/infinispan/device/CredentialKey.java]: You are right, I think I need to re-add them. I got an issue with this atm.

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

File: pkg/controller/iotproject/tenant_cleanup.go (hunk 4 above)
Comment (ctron) [reply pkg/controller/iotproject/tenant_cleanup.go]: It actually is … however it must be stable. The idea is that different names, with different namespaces result in different hashes. so NS1:"foo" gives a different named UUID than NS2:"foo".

Comment (ctron) [reply pkg/controller/iotproject/tenant_cleanup.go]: It actually is … however it must be stable. The idea is that different names, with different namespaces result in different hashes. so NS1:"foo" gives a different named UUID than NS2:"foo".

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

File: iot/iot-device-registry-infinispan/src/test/java/io/enmasse/iot/registry/infinispan/device/AbstractCredentialsManagementServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.enmasse.iot.infinispan.device.DeviceKey;
+import io.opentracing.Span;
+import io.opentracing.noop.NoopSpan;
+
+public class AbstractCredentialsManagementServiceTest {
```
[the remark this answers is not in the document]
Comment (ctron) [reply iot/iot-device-registry-infinispan/src/test/java/io/enmasse/iot/registry/infinispan/device/AbstractCredentialsManagementServiceTest.java]: I added a comment to explain.

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

File: pkg/controller/iotproject/tenant_cleanup.go (hunk 4 above)
Comment (dejanb) [reply pkg/controller/iotproject/tenant_cleanup.go]: Sounds good. Was just wondering about why that specific value.

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot

**Comment by kornys** [comment]
@enmasse-ci run tests profile=iot

**Comment by kornys** [comment]
@enmasse-ci run tests profile=iot

**Comment by ctron** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.isolated.registry.InfinispanDeviceRegistryTest

**Pull request merged**